File: src/app/dto/JobDTO.ts
import { CompanyDTO } from './CompanyDTO';
import { DTO } from './DTO';
import { EmployeeDTO } from './EmployeeDTO';

export class JobDTO implements DTO {
  id!: number;
  employee!: EmployeeDTO;
  company!: CompanyDTO;
  function!: string;
  beginDate!: Date;
  endDate!: Date;

  constructor() {
    this.employee = new EmployeeDTO;
    this.company = new CompanyDTO;
    this.beginDate = new Date;
    this.endDate = new Date;
  }
}

File: src/main/java/teampearl/cvgenerator/back_end/presentation_layer/dto/JobDTO.java
package teampearl.cvgenerator.back_end.presentation_layer.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import teampearl.cvgenerator.back_end.data_access_layer.entities.Company;
import teampearl.cvgenerator.back_end.data_access_layer.entities.Employee;

import java.time.LocalDate;


public class JobDTO extends Participation{
    @JsonProperty("id")
    private int id;

    @JsonProperty("company")
    private CompanyDTO company;

    @JsonProperty("employee")
    private EmployeeDTO employee;

    @JsonProperty("function")
    private String function;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public CompanyDTO getCompany() {
        return company;
    }

    public void setCompany(CompanyDTO company) {
        this.company = company;
    }

    public EmployeeDTO getEmployee() {
        return employee;
    }

    public void setEmployee(EmployeeDTO employee) {
        this.employee = employee;
    }

    public String getFunction() {
        return function;
    }

    public void setFunction(String function) {
        this.function = function;
    }
}
